native/src/main/java/io/ballerina/stdlib/sql/datasource/SQLDatasource.java

```diff
@@ -320,23 +321,31 @@ private HikariDataSource buildNonXADataSource(SQLDatasourceParams sqlDatasourceP
             if (sqlDatasourceParams.connectionPool != null) {
                 int maxOpenConn = sqlDatasourceParams.connectionPool.
                         getIntValue(Constants.ConnectionPool.MAX_OPEN_CONNECTIONS).intValue();
-                if (maxOpenConn > 0) {
+                if (maxOpenConn >= 1) {
                     config.setMaximumPoolSize(maxOpenConn);
+                } else {
+                    throw new ApplicationError("ConnectionPool field 'maxOpenConnections' cannot be less than one.");
```

Comment:
```suggestion
                if (maxOpenConn < 1) {
                    throw new ApplicationError("ConnectionPool field 'maxOpenConnections' cannot be less than one.");
                }
                config.setMaximumPoolSize(maxOpenConn);
```

Would this be cleaner code? Can get rid of the else block

Reply:
Fixed in c101064